### PR TITLE
add doc for array fix:#10

### LINKED ISSUE #10: array/: add doc
go doc guide:

https://blog.golang.org/godoc-documenting-go-code

- [ ] Document package, types, public function/method/variables.

### PR DESCRIPTION
**Pull request by pengsven**
# Description

Add doc for array

Fixes #10 

## Type of change

- [ ] Document changes

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: array/compact.go [hunk 1]
```diff
@@ -49,6 +51,9 @@ func (sa *CompactedArray) appendElt(index uint32, elt []byte) {
 var ErrIndexLen = errors.New("the length of index and elts must be equal")
 var ErrIndexNotAscending = errors.New("index must be an ascending slice")
 
+// Init initializes a compacted array from the slice type elts
+// the index parameter must be a non-descending array of type unit32,
```
Comment (liubaohai) [inline array/compact.go]: `non-descending` 代表升序的意思吗？如果是，可以直接用`ascending`

File: array/converter.go [hunk 2]
```diff
@@ -55,6 +57,7 @@ func (c U32Conv) GetMarshaledEltSize(b []byte) uint32 {
 	return 4
 }
 
+// ByteConv is a EltConverter is used to convert []byte between serialized byte stream
```
Comment (liubaohai) [inline array/converter.go]: `convert between serialized byte stream and []byte`感觉这样通顺点。

File: array/converter.go [hunk 3]
```diff
@@ -7,15 +7,16 @@ import (
 // A EltConverter is used to convert one element between serialized byte stream
 // and in-memory data structure
 type EltConverter interface {
-	MarshalElt(interface{}) []byte
-	UnmarshalElt([]byte) (uint32, interface{})
+	MarshalElt(interface{}) []byte             // convert element to byte stream
```
Comment (drmingdrmer) [inline array/converter.go]: hm...感觉这个借口里的`Elt`都有点多余...后面把它去掉咋样?

File: array/example_compact_test.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package array
+
+import (
+	"fmt"
+)
+
+func Example() {
```
Comment (drmingdrmer) [inline array/example_compact_test.go]: func name好像应该是Examplewhatwhat 这样: ExampleInitGet酱

File: array/example_compact_test.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	indexes := []uint32{0, 5, 9, 203, 400}
+	eltsData := []uint32{12, 15, 19, 120, 300}
+
+	ca := CompactedArray{EltConverter: U32Conv{}}
+	err := ca.Init(indexes, eltsData)
+	if err != nil {
+		fmt.Printf("Init compacted array error:%s\n", err)
+		return
```
Comment (drmingdrmer) [inline array/example_compact_test.go]: 在example貌似不需要return, 最后需要加上注释`Output: foo bar`表示它输出的是什么, 运行测试时会自动检查Output里的内容.

File: array/converter.go (hunk 3 above)
Comment (pengsven) [reply array/converter.go]: 1

**Pull request merged**